src/django/VLE/views/assignment.py

```
@@ -188,6 +188,14 @@ def partial_update(self, request, *args, **kwargs):
             if 'lti_id' in req_data:
                 factory.make_lti_ids(lti_id=req_data['lti_id'], for_model=Lti_ids.ASSIGNMENT, assignment=assignment)
 
+            # If a entry has been submitted to one of the journals of the journal it cannot be unpublished
+            if assignment.is_published and 'is_published' in req_data and not req_data['is_published'] and \
+               Entry.objects.filter(node__journal__assignment=assignment).exists():
+                req_data['is_published'] = True
```

I feel this should throw a bad_request instead, so the user is informed of their malformed attempt, instead of silently failing.